app/presenters/tree_node_builder.rb

```diff
@@ -1,6 +1,8 @@
 class TreeNodeBuilder
   include MiqAeClassHelper
 
+  @mutex = Mutex.new
```

Comment:
why not hide this in a class method like:
```ruby
def mutex
  @mutex ||= Mutex.new
end
```

and have the caller call `mutex.syncronize` instead of `@mutex.syncronize`